src/Core/src/Result.php

```diff
@@ -62,6 +62,32 @@ public function __destruct()
         return $this->response->resolve($timeout);
     }
 
+    /**
+     * Make sure all provided requests are executed.
+     *
+     * @param self[]     $results
+     * @param float|null $timeout Duration in seconds before aborting. When null wait
+     *                            until the end of execution. Using 0 means non-blocking
+     *
+     * @return iterable<self>
+     *
+     * @throws NetworkException
+     * @throws HttpException
+     */
+    final public static function multiplex(iterable $results, float $timeout = null): iterable
```

Comment:
You suggested `resolveAll()` in a comment before. I like that better.

Reply:
issue with `resolveAll`, is: it sounds like `resolve`, people may expect the same behavior:
- when yielding true => request resolve in given timeout
- when yielding false => request not resolved in given timeout
- What if exception occures?.. We can't yield an exception and continue processing other reponse.

any idea to solve this?

Reply:
Hm.. lets make sure we dont throw anything... Ie, just name this `download()`. 

```php
while(!empty/$results) {
  Result::download($results, 2.0);
  foreach ($results as $i => $result) {
    if ($result->info()['resolved']) {
      unset($results[$i];
      $result->getPayload(); // This may throw exception. 
    }
  }
}

```

Reply:
This example is no good. See https://github.com/async-aws/aws/pull/417#issuecomment-606894093